contracts/src/CW721ERC721Pointer.sol

```diff
@@ -112,6 +116,31 @@ contract CW721ERC721Pointer is ERC721,ERC2981 {
         return (AddrPrecompile.getEvmAddr(string(addr)), amt);
     }
 
+    function tokenURI(uint256 tokenId) public view override returns (string memory) {
+        // revert if token isn't owned
+        ownerOf(tokenId);
+        string memory tId = _formatPayload("token_id", _doubleQuotes(Strings.toString(tokenId)));
+        string memory req = _curlyBrace(_formatPayload("nft_info", _curlyBrace(tId)));
+        bytes memory response = WasmdPrecompile.query(Cw721Address, bytes(req));
+        bytes memory uri = JsonPrecompile.extractAsBytes(response, "token_uri");
+        return string(uri);
+    }
+
+    // 721-Enumerable
+    function totalSupply() public view virtual returns (uint256) {
```

Comment:
what happens for nft collections that have unlimited supply? are they then not supported by pointers?

Reply:
`totalSupply` returns the total amount of tokens stored by the contract. As tokens get minted, this value will increase; as tokens get burned, this value will decrease. Collections with uncapped token supply will still be supported.

Reference: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/IERC721Enumerable.sol

Reply:
i see that makes sense, i mistook `totalSupply` for the max supply of the collection